src/Domain/Creators/Sphere.cpp

```diff
@@ -377,14 +431,24 @@ Sphere::external_boundary_conditions() const {
   }
   return boundary_conditions;
 }
+
 std::unordered_map<std::string,
                    std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
 Sphere::functions_of_time(const std::unordered_map<std::string, double>&
                               initial_expiration_times) const {
-  if (time_dependence_->is_none()) {
-    return {};
+  if (time_dependent_options_.has_value()) {
+    return std::visit(
+        [&initial_expiration_times](const auto& arg) {
+          using T = std::decay_t<decltype(arg)>;
+          if constexpr (std::is_same_v<T, sphere::TimeDependentMapOptions>) {
+            return arg.create_functions_of_time(initial_expiration_times);
+          } else {
+            return arg->functions_of_time(initial_expiration_times);
+          }
+        },
+        time_dependent_options_.value());
```

Comment:
[optional] If you're just going to branch on the type anyway, it's I think it's cleaner to skip the visit and just do the access directly:
```c++
if (std::holds_alternative<sphere::TimeDependentMapOptions>(
        time_dependent_options_.value())) {
  return std::get<sphere::TimeDependentMapOptions>(
      time_dependent_options_.value())
      .create_functions_of_time(initial_expiration_times);
} else {
  return std::get<
      std::unique_ptr<domain::creators::time_dependence::TimeDependence<3>>>(
          time_dependent_options_.value())
          ->functions_of_time(initial_expiration_times);
}
```
(Type aliases and local reference variables might make this more readable.)